packages/client/src/utils/uploadScreenshot.ts

```diff
@@ -0,0 +1,67 @@
+import fs from "fs";
+import FormData from "form-data";
+import mime from "mime";
+import path from "path";
+import { UploadParameters } from "../models/UploadParameters";
+import { snakelize } from "./str";
+import uniqid from "uniqid";
+
+function buildScreenshotKey(
+  fileName: string,
+  repository: string,
+  sliceName: UploadParameters["sliceName"],
+  variationId: UploadParameters["variationId"]
+): string {
+  return `${repository}/shared-slices/${snakelize(sliceName)}/${snakelize(
+    variationId
+  )}-${uniqid()}/${fileName}`;
+}
+
+export function uploadScreenshot({
+  acl,
+  repository,
+  sliceName,
+  variationId,
+  filePath,
+}: UploadParameters & { repository: string }): Promise<string> {
+  const form = new FormData();
+  const fileName = path.basename(filePath);
+
+  // adding fields
+  Object.entries(acl.fields).forEach(([key, value]) => {
+    form.append(key, value);
+  });
+
+  // building then adding the key
+  const screenshotKey = buildScreenshotKey(
+    fileName,
+    repository,
+    sliceName,
+    variationId
+  );
+  form.append("key", screenshotKey);
+
+  // adding content type
+  const fileExtension = fileName.split(".").pop();
+  form.append("Content-Type", fileExtension && mime.getType(fileExtension));
+
+  // adding the file
+  form.append("file", fs.createReadStream(filePath), {
```

Comment:
this can fail because s3 needs to know the file length in-order to upload the screenshot, there's a solution for this in the init

Reply:
### Stage Repository
<img width="687" alt="image" src="https://user-images.githubusercontent.com/41187899/174310982-32ce573a-5b3f-4cde-969a-d17d8df58660.png">
<img width="1131" alt="image" src="https://user-images.githubusercontent.com/41187899/174311078-dffdcba4-0087-4738-8188-42cb3e6838f5.png">

### Production Repository
<img width="687" alt="image" src="https://user-images.githubusercontent.com/41187899/174311835-775e63d6-3729-4b79-864b-65dd6b0582ef.png">
<img width="1141" alt="image" src="https://user-images.githubusercontent.com/41187899/174311935-447d141f-544e-4b4c-9681-f454e6bdbd1f.png">

Reply:
Here's the old version of the upload: https://github.com/prismicio/slice-machine/blob/master/packages/slice-machine/lib/models/common/http/upload.ts

As you can see, I still send data the same way, i don't think I modified anything (especially because I don't know about S3 contracts haha 😂 )